components/lark-type-check/src/hir_typeck.rs

```diff
@@ -423,7 +429,7 @@ where
                         EntityData::LangItem(LangItem::Uint) => uint_type,
                         EntityData::Error(_) => self.error_type(),
                         _ => {
-                            self.record_error(expression);
+                            self.record_error("unknown type for expression".into(), expression);
```

Comment:
the idea here was that (for now, anyway) operators like `+` etc require integers. So perhaps a better message might be "can't apply `+` to a `String`" or something? (meh, whatever)

Reply:
Sure, will take a look